### PR TITLE
Fix and improve metrics for ServiceCatalog

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Fixed metrics for ServiceCatalog

**Related issue(s)**

### CODE REVIEW

File: resources/service-catalog/charts/catalog-crds/templates/metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    port: secure
+    scheme: https
+    tlsConfig:
+      insecureSkipVerify: true
+  namespaceSelector:
+    matchNames:
+    - "{{ .Release.Namespace }}"
+  selector:
+    matchLabels:
+      app: {{ template "fullname" . }}-controller-manager
```
Comment (hisarbalik) [inline resources/service-catalog/charts/catalog-crds/templates/metrics.yaml]: Service definition above should match this selectors? Or do you have some where else another service definition match this monitoring?

File: resources/service-catalog/charts/catalog-crds/templates/metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    prometheus: monitoring
+    app: {{ template "fullname" . }}-controller-manager
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+  name: {{ template "fullname" . }}-controller-manager
+spec:
+  endpoints:
+  - interval: 30s
+    port: secure
```
Comment (hisarbalik) [inline resources/service-catalog/charts/catalog-crds/templates/metrics.yaml]: is the port for service definition above?

Comment (hisarbalik) [reply resources/service-catalog/charts/catalog-crds/templates/metrics.yaml]: if yes you should change port name

File: resources/service-catalog/charts/catalog-crds/templates/metrics.yaml (hunk 1 above)
Comment (polskikiel) [reply resources/service-catalog/charts/catalog-crds/templates/metrics.yaml]: I have another service definition which matches this selector in this file:
 ```kyma/resources/service-catalog/charts/catalog-crds/templates/controller-manager-service.yaml```

File: resources/service-catalog/charts/catalog-crds/templates/metrics.yaml (hunk 2 above)
Comment (polskikiel) [reply resources/service-catalog/charts/catalog-crds/templates/metrics.yaml]: It's a port from the file: ```kyma/resources/service-catalog/charts/catalog-crds/templates/controller-manager-service.yaml```. I think this service definition is now unused so I will delete it 👍

**Pull request merged**